source/core/parameters.cc

```diff
@@ -2378,32 +2378,43 @@ namespace Parameters
                           "State whether MG should print max and min eigenvalue"
                           "Choices are <quiet|verbose>.");
 
-        prm.declare_entry("mg coarse grid max iterations",
+        prm.declare_entry("mg coarse grid solver",
+                          "gmres",
+                          Patterns::Selection("gmres|amg|ilu|direct"),
+                          "The coarse grid solver for lsmg or gcmg"
+                          "Choices are <gmres|amg|ilu|direct>.");
+
+        prm.declare_entry("mg gmres max iterations",
                           "2000",
                           Patterns::Integer(),
-                          "mg coarse grid iterations for lsmg or gcmg");
+                          "mg gmres iterations for lsmg or gcmg");
 
-        prm.declare_entry("mg coarse grid tolerance",
+        prm.declare_entry("mg gmres tolerance",
                           "1e-14",
                           Patterns::Double(),
-                          "mg coarse grid tolerance n for lsmg or gcmg");
+                          "mg gmres tolerance n for lsmg or gcmg");
 
-        prm.declare_entry("mg coarse grid reduce",
+        prm.declare_entry("mg gmres reduce",
                           "1e-4",
                           Patterns::Double(),
-                          "mg coarse grid reduce for lsmg or gcmg");
+                          "mg gmres reduce for lsmg or gcmg");
 
-        prm.declare_entry("mg coarse grid max krylov vectors",
+        prm.declare_entry("mg gmres max krylov vectors",
                           "30",
                           Patterns::Integer(),
-                          "mg coarse grid max krylov vectors for lsmg or gcmg");
+                          "mg gmres max krylov vectors for lsmg or gcmg");
 
-        prm.declare_entry("mg coarse grid preconditioner",
+        prm.declare_entry("mg gmres preconditioner",
                           "amg",
                           Patterns::Selection("amg|ilu"),
-                          "The preconditioner for the mg coarse grid solver"
+                          "The preconditioner for the mg gmres solver"
                           "Choices are <amg|ilu>.");
 
+        prm.declare_entry("mg use amg default parameters",
```

Comment:
I think this is aligned with the other names:
```suggestion
        prm.declare_entry("mg amg use default parameters",
```

Reply:
agreed